### PR TITLE
Fix client configured hashing

### PR DESCRIPTION
**Pull request by jaschrep-msft**
Fixes files.shares transfer validation tests and the bug they were hiding

### CODE REVIEW

**Review by amnguye — APPROVED** [review]

**Review by seanmcc-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
@jaschrep-msft, can you update the Files changelog?

File: sdk/storage/Azure.Storage.Common/src/Shared/TransferValidationOptionsExtensions.cs [hunk 1]
```diff
@@ -9,7 +9,11 @@ public static ValidationAlgorithm ResolveAuto(this ValidationAlgorithm validatio
         {
             if (validationAlgorithm == ValidationAlgorithm.Auto)
             {
+#if BlobSDK || DataLakeSDK
                 return ValidationAlgorithm.StorageCrc64;
+#elif FileSDK // file shares don't support crc64
+                return ValidationAlgorithm.MD5;
+#endif
```
Comment (jaschrep-msft) [inline sdk/storage/Azure.Storage.Common/src/Shared/TransferValidationOptionsExtensions.cs:16]: Should probably include `#else throw` in here for safety

Comment (jaschrep-msft) [reply sdk/storage/Azure.Storage.Common/src/Shared/TransferValidationOptionsExtensions.cs:16]: > @jaschrep-msft, can you update the Files changelog?

@seanmcc-msft The bugged code hasn't actually been released yet. There would be no change to alert someone to.

**Comment by jaschrep-msft** [comment]
> @jaschrep-msft, can you update the Files changelog?

@seanmcc-msft Bugged code hasn't been released. There's no change to alert developers to.

**Review by amnguye — APPROVED** [review]

**Pull request merged**